src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -181,18 +194,26 @@ impl TimeTrigger {
             let now: std::time::Duration = SystemTime::now()
                 .duration_since(UNIX_EPOCH)
                 .expect("system time before Unix epoch");
-            NaiveDateTime::from_timestamp_opt(now.as_secs() as i64, now.subsec_nanos())
+            DateTime::from_timestamp(now.as_secs() as i64, now.subsec_nanos())
                 .unwrap()
+                .naive_local()
                 .and_local_timezone(Local)
                 .unwrap()
         };
 
         #[cfg(not(test))]
         let current = Local::now();
-        let next_time = TimeTrigger::get_next_time(current, config.interval, config.modulate);
+        // In the case where bad user input results in an invalid next time, provide a valid time.
+        let next_time = TimeTrigger::get_next_time(current, config.interval, config.modulate)
+            .unwrap_or(current + Duration::try_seconds(1_i64).unwrap());
```

Comment:
We probably want to tell the user if get_next_time fails right?

Reply:
The only solution I see to do that is to panic (aka just unwrap without the or). I'm ok with that; if I'm interpreting you correctly?

Reply:
We can just use `unwrap` here, and in #347 , I will call `get_next_time` in `trigger`, which can handle the error